openshift/openshift/src/main/java/org/arquillian/cube/openshift/impl/client/OpenShiftAssistant.java

```diff
@@ -181,4 +179,62 @@ public OpenShiftAssistantTemplate usingTemplate(URL templateURL) {
     public OpenShiftAssistantTemplate usingTemplate(String templateURL) throws MalformedURLException {
         return new OpenShiftAssistantTemplate(new URL(templateURL), getClient());
     }
+
+    /**
+     * Gets the list of all the OpenShift Projects.
+     *
+     * @return list of OpenShift Projects.
+     */
+    public List<Project> listProjects() {
+        return getClient().projects().list().getItems();
+    }
+
+    /**
+     * Gets the current OpenShift project used for deploying application.
+     *
+     * @return current namespace.
+     */
+    public String getCurrentProject() {
+        return getClient().getNamespace();
+    }
+
+    /**
+     * Checks if the given project exists or not.
+     *
+     * @param name project name
+     * @return true/false
+     * @throws IllegalArgumentException
+     */
+    public boolean projectExists(String name) throws IllegalArgumentException {
+        if (name == null || name.isEmpty()) {
+            throw new IllegalArgumentException("Project name cannot be empty");
+        }
+        return getClient().projects().list().getItems().stream()
+            .map(p -> p.getMetadata().getName())
+            .anyMatch(Predicate.isEqual(name));
+    }
+
+    /**
+     * Finds for the given project.
+     *
+     * @param name project name
+     * @return given project or null if project does not exist
+     * @throws IllegalArgumentException
+     */
+    public Optional<Project> findProject(String name) throws IllegalArgumentException {
+        if (name == null || name.isEmpty()) {
+            throw new IllegalArgumentException("Project name cannot be empty");
+        }
+        return projectExists(name) ? getProject(name) : null;
```

Comment:
This return can be simplified to `getProject(name)` and when you do the `findFirst` it automatically takes care of not finding the element, so it is not necessary to call `projectExists`